src/main/java/io/codelex/flightplanner/service/FlightDatabaseService.java

```diff
@@ -0,0 +1,118 @@
+package io.codelex.flightplanner.service;
+
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import io.codelex.flightplanner.repository.AirportRepository;
+import io.codelex.flightplanner.repository.FlightRepository;
+import io.codelex.flightplanner.request.AddFlightRequest;
+import io.codelex.flightplanner.request.PageResult;
+import io.codelex.flightplanner.request.SearchFlightsRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class FlightDatabaseService implements FlightService {
+
+    private final FlightRepository flightRepository;
+    private final AirportRepository airportRepository;
+
+    public FlightDatabaseService(FlightRepository flightRepository, AirportRepository airportRepository) {
+        this.flightRepository = flightRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    @Override
+    public Flight addFlight(AddFlightRequest request) {
+
+        String departureAirport = request.getFrom().getAirport().trim();
+        String arrivalAirport = request.getTo().getAirport().trim();
+        String departureCity = request.getFrom().getCity().trim();
+        String arrivalCity = request.getTo().getCity().trim();
+        String departureCountry = request.getFrom().getCountry().trim();
+        String arrivalCountry = request.getTo().getCountry().trim();
+
+        if (departureAirport.equalsIgnoreCase(arrivalAirport) &&
```

Comment:
Visticamāk šo pārbaudi iznestu citā metodē, lai šo saīsinātu - pati metode var saukties piemēram validateAirports un viņa var mest kļūdu, ja tās ir vienādas.

Tas pats ar departure/arrival time - arī šo pārbaudi var iznest atsevišķā metodē.

Kā arī to existing flights pārbaudi arī iznes atsevišķā metodē - kods uzreiz tīrāks izskatīsies.